operator/pkg/manifest/client.go

```diff
@@ -142,16 +141,38 @@ func (h *HelmClient) SetFlags(args map[string]map[string]interface{}, actionClie
 		if !value.IsValid() || !value.CanSet() {
 			continue
 		}
+
+		validConversion := true
+
 		//nolint:exhaustive
 		switch value.Kind() {
 		case reflect.Bool:
-			value.SetBool(flagValue.(bool))
-		case reflect.Int:
-			value.SetInt(flagValue.(int64))
-		case reflect.Int64:
-			value.SetInt(flagValue.(int64))
+			var valueToBeSet bool
+			valueToBeSet, validConversion = flagValue.(bool)
+			if validConversion {
+				value.SetBool(valueToBeSet)
+			}
+		case reflect.Int, reflect.Int64:
+			var valueToBeSet int64
+			valueToBeSet, validConversion = flagValue.(int64)
+			if validConversion {
+				value.SetInt(valueToBeSet)
+			}
+			var fallbackInt64 time.Duration
+			fallbackInt64, validConversion = flagValue.(time.Duration)
```

Comment:
no matter previous type assertion is success or not, if the type assertion here is failed, the `validConversion` will overwrite to `false`, I think that's not what you want, right?
And can we extract this switch case to a function and write test for it, it make sense to see some test case.